pkg/controller/clusterstorage/clusterstorage_controller.go

```diff
@@ -186,7 +192,58 @@ func (r *ReconcileClusterStorage) Reconcile(request reconcile.Request) (reconcil
 	return reconcile.Result{}, nil
 }
 
-// syncStatus will set either Available=true;Failing=false;Progressing=false;
+var (
+	unavailable = configv1.ClusterOperatorStatusCondition{
+		Type:   configv1.OperatorAvailable,
+		Status: configv1.ConditionFalse,
+	}
+	notFailing = configv1.ClusterOperatorStatusCondition{
+		Type:   configv1.OperatorFailing,
+		Status: configv1.ConditionFalse,
+	}
+	notProgressing = configv1.ClusterOperatorStatusCondition{
+		Type:   configv1.OperatorProgressing,
+		Status: configv1.ConditionFalse,
+	}
+)
+
+// setStatusProgressing sets Available=false;Failing=false;Progressing=true
+// we set "progressing" if the cluster operator's version is not the latest
+// and we are about to try to roll it out
+func (r *ReconcileClusterStorage) setStatusProgressing(clusterOperator *configv1.ClusterOperator) error {
```

Comment:
Can we write some unit tests for this stuff?

Reply:
done. will write one for syncStatus in follow up